test/test_instructions.py

```diff
@@ -1227,6 +1216,127 @@ def test_sign_multiple_operation(
+
+    account = DEFAULT_ACCOUNT
+    snap_path = Path(f"{account}")
+
+    tezos_navigator.disable_hwm(snap_path)
+
+    tezos_navigator.setup_app_context(
+        account,
+        DEFAULT_CHAIN_ID,  # Chain = 0
+        main_hwm=Hwm(0, 0),
+        test_hwm=Hwm(0, 0)
+    )
+
+    attestation = build_attestation(
+        1, 0,
+        DEFAULT_CHAIN_ID  # Chain = 0
+    )
+
+    client.sign_message(account, attestation)
+
+    tezos_navigator.check_app_context(
+        account,
+        chain_id=DEFAULT_CHAIN_ID,
+        main_hwm=Hwm(1, 0),
+        test_hwm=Hwm(0, 0),
+        snap_path=Path("sign_1_0"))
+
+    attestation = build_attestation(
+        2, 0,
+        "NetXH12AexHqTQa"  # Chain = 1
+    )
+
+    client.sign_message(account, attestation)
+
+    tezos_navigator.check_app_context(
+        account,
+        chain_id=DEFAULT_CHAIN_ID,
+        main_hwm=Hwm(2, 0),
+        test_hwm=Hwm(0, 0),
+        snap_path=Path("sign_2_0"))
+
+    attestation = build_attestation(
+        2, 0,
+        "NetXH12Af5mrXhq"  # Chain = 2
+    )
+
+    with StatusCode.WRONG_VALUES.expected():
+        client.sign_message(account, attestation)
+
+    tezos_navigator.check_app_context(
+        account,
+        chain_id=DEFAULT_CHAIN_ID,
+        main_hwm=Hwm(2, 0),
+        test_hwm=Hwm(0, 0),
+        snap_path=Path("sign_2_0")
+    )
+
+
+def test_hwm_disabled_manual(client: TezosClient, tezos_navigator: TezosNavigator, backend_name) -> None:
+    """On device test to verify HWM settings operation. Can run the test with hwm setting enabled or disabled.
+       When HWM is disabled, an abrupt power off will result in HWM reset to 0.
+       Whwereas when HWM settings is enabled, the abrupt power off will not
+       reset the HWM."""
+    # check if backend is speculos, then return .
+    if backend_name == "speculos":
+        assert True
+        return
+
+    account = DEFAULT_ACCOUNT
+    snap_path = Path(f"{account}")
+
+    hwm_input = input("Is hwm disabled ? (y/n)")
+    hwm_disabled = False
+    if hwm_input.find("y") != -1:
+        hwm_disabled = True
+
+    tezos_navigator.setup_app_context(
+        account,
+        DEFAULT_CHAIN_ID,  # Chain = 0
+        main_hwm = Hwm(0, 0),
+        test_hwm = Hwm(0, 0)
+    )
+
+    for i in range(1, 11):
+        attestation = build_attestation(i, 0, DEFAULT_CHAIN_ID)
+        client.sign_message(account, attestation)
+
+    tezos_navigator.check_app_context(
+        account,
+        DEFAULT_CHAIN_ID,
+        main_hwm = Hwm(10,0),
+        test_hwm = Hwm(0,0),
+        snap_path = snap_path,
+        backend_name = backend_name
+    )
+    output = "No"
+    while output.find("Y") == -1 and output.find("y") == -1:
+        output = input("1. Now switch off the device abruptly. \n"
+                       "2. Reconnect it to power source. \n"
+                       "3. Open the baking app."
+                       "Done? (Yes/No)")
```

Comment:
Does quitting the application normally save the `HWM`?

Reply:
Yes

Reply:
Can you also add this case in this test ?
```
If HWM disable + quitting app normally, then HWM saved.
```

Reply:
Added this test.